data_validation/partition_builder.py

```diff
@@ -0,0 +1,241 @@
+
+            # Get Source and Target Primary key Max
+            source_max_query = source_partition_row_builder.get_max_query()
+            source_max = source_max_query.execute()
+
+            target_max_query = target_partition_row_builder.get_max_query()
+            target_max = target_max_query.execute()
+
+            if source_min != target_min:
+                logging.warning(
+                    "max(partition_key) for Source and Target tables do not"
+                    "match, proceeding with max(source_max, target_max)"
+                )
+
+            upper_bound = max(source_max, target_max)
+
+            filter_list = []  # Store partition filters per config/table
+            i = 0
+            marker = lower_bound
+            partition_step = (upper_bound - lower_bound) // partition_count
+            while i < partition_count:
+                lower_val = marker
+                upper_val = marker + partition_step
+
+                if i == partition_count - 1:
+                    upper_val = upper_bound + 1
+
+                partition_filter = (
+                    f"{self.partition_key} >= {lower_val} "
+                    f"and {self.partition_key} < {upper_val}"
+                )
+                filter_list.append(partition_filter)
+
+                i += 1
+                marker += partition_step
+
+            master_filter_list.append(filter_list)
+
+        return master_filter_list
+
+    def _add_partition_filters_and_store(
+        self,
+        partition_filters: List[List[str]],
+    ) -> None:
+        """Add Partition Filters to ConfigManager and return a list of ConfigManager objects.
+
+        Args:
+            partition_filters (List[List[str]]): List of List of Partition filters
+            for all Table/ConfigManager objects
+
+        Returns:
+            None
+        """
+
+        table_count = len(self.config_managers)
+        for ind in range(table_count):
+            config_manager = self.config_managers[ind]
+            filter_list = partition_filters[ind]
+
+            target_folder_name = config_manager.full_source_table
+            target_folder_path = cli_tools.get_target_table_folder_path(
+                self.config_dir, target_folder_name
+            )
+
+            for pos in range(len(filter_list)):
+                filter_dict = {
+                    "type": "custom",
+                    "source": filter_list[pos],
+                    "target": filter_list[pos],
+                }
+                # Append partition new filter
+                config_manager.filters.append(filter_dict)
+
+                # Save partition
+                yaml_configs = self._get_yaml_from_config(config_manager)
+                target_file_name = "0" * (4 - len(str(pos))) + str(pos) + ".yaml"
+                config_file_path = os.path.join(target_folder_path, target_file_name)
+                cli_tools.store_validation(config_file_path, yaml_configs)
```

Comment:
Each store_validation call logs a success message for the file being written [here](https://github.com/GoogleCloudPlatform/professional-services-data-validator/blob/develop/data_validation/state_manager.py#L175) - if a user has 1000 partitions this would log a lot of times... did you encounter this while testing?